File: fe/ClinicManager/src/models/app_settings.ts
export interface AppSettings {
    id: number;
    voucherExpiryTime: number;
    visitAbsenceRate: number;
    visitVipRate: number;
    boostNetRate: number;
    googleReviewDiscount: number;
    booksyHappyHours: number;
}

export interface NewAppSettings {
    voucherExpiryTime: number;
    visitAbsenceRate: number;
    visitVipRate: number;
    boostNetRate: number;
    googleReviewDiscount: number;
    booksyHappyHours: number;
}

export interface DiscountSettings {
    googleReviewDiscount: number;
    booksyHappyHours: number;
}

File: be/ClinicManager/src/main/java/com/clinic/clinicmanager/model/AppSettings.java
package com.clinic.clinicmanager.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "app_settings")
public class AppSettings {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    @Builder.Default
    private Integer voucherExpiryTime = 3; //MONTHS

    @Column(nullable = false)
    @Builder.Default
    private Integer visitAbsenceRate = 80; //%

    @Column(nullable = false)
    @Builder.Default
    private Integer visitVipRate = 140; //%

    @Column(nullable = false)
    @Builder.Default
    private Integer boostNetRate = 45; //%

    @Column(nullable = false)
    @Builder.Default
    private Integer googleReviewDiscount = 5; //%

    @Column(nullable = false)
    @Builder.Default
    private Integer booksyHappyHours = 5; //%
}
